File: app/src/main/java/vn/loitp/app/activity/demo/fragmentnavigation/FN1.java
package vn.loitp.app.activity.demo.fragmentnavigation;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.core.base.BaseFragment;
import com.core.utilities.LLog;

import loitp.basemaster.R;

public class FN1 extends BaseFragment implements OnBackPressedListener {
    private FragmentNavigationActivity fragmentNavigationActivity;

    public FN1() {

    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        fragmentNavigationActivity = (FragmentNavigationActivity) getActivity();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.fn_1;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        final Button button = view.findViewById(R.id.button);
        button.setOnClickListener(v -> {
            fragmentNavigationActivity.getNavController().navigate(R.id.action_fn1_to_fn2);
        });
    }

    @Override
    public void onResume() {
        super.onResume();
        LLog.d(fragmentNavigationActivity.T, "onResume FN1");
    }

    @Override
    public void onBackPressed() {
        LLog.d(fragmentNavigationActivity.T, "onBackPressed FN1");
        fragmentNavigationActivity.popThisFragment();
    }
}

File: app/src/main/java/vn/loitp/app/activity/demo/fragmentNavigation/FN1.kt
package vn.loitp.app.activity.demo.fragmentNavigation

import android.os.Bundle
import android.view.View
import com.loitp.annotation.LogTag
import com.loitp.core.base.BaseFragment
import com.loitp.core.base.OnBackPressedListener
import kotlinx.android.synthetic.main.frm_fn_1.*
import vn.loitp.app.R

@LogTag("fragmentNavigationActivity")
class FN1 : BaseFragment(), OnBackPressedListener {

    private var fragmentNavigationActivity: FragmentNavigationActivity? = null

    override fun setLayoutResourceId(): Int {
        return R.layout.frm_fn_1
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        fragmentNavigationActivity = activity as FragmentNavigationActivity?
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        button.setOnClickListener {
            fragmentNavigationActivity?.navController?.navigate(R.id.action_fn1_to_fn2)
        }
        view.findViewById<View>(R.id.button4)
            .setOnClickListener { fragmentNavigationActivity?.navController?.navigate(R.id.action_fn1_to_fn4) }
    }

    override fun onResume() {
        super.onResume()
        logD("onResume FN1")
    }

    override fun onBackPressed(): Boolean {
        logD("onBackPressed FN1")
        fragmentNavigationActivity?.popThisFragment()
        return true
    }
}
